sherpa-onnx/csrc/online-transducer-modified-beam-search-decoder.cc

```diff
@@ -129,6 +129,22 @@ void OnlineTransducerModifiedBeamSearchDecoder::Decode(
         model_->RunJoiner(std::move(cur_encoder_out), View(&decoder_out));
 
     float *p_logit = logit.GetTensorMutableData<float>();
+
+    // copy raw logits, apply temperature-scaling (for confidences)
+    int32_t p_logit_items = vocab_size * num_hyps;
+    std::vector<float> logit_with_temperature(p_logit_items);
```

Comment:
Can we change p_logit in-place?

Reply:
no, it cannot be done in-place

the idea is to apply temperature only for computation of confidences, 
the decoding continues to use the original values

this is why the logit values are copied to a new buffer

Reply:
Thanks for the explanation.